engine/src/main.cpp

```diff
@@ -371,13 +371,6 @@ int main(int argc, char *argv[]) {
 #endif
 
     std::vector<std::vector<char> > temp = ROLES::getAllRolePriorities();
-#if defined(HAVE_SDL)
-#ifndef NO_SDL_JOYSTICK
-    if (SDL_InitSubSystem(SDL_INIT_JOYSTICK)) {
-        VS_LOG_FLUSH_EXIT(fatal, (boost::format("Couldn't initialize SDL: %1%") % SDL_GetError()), 1);
-    }
-#endif
-#endif
```

Comment:
we always have SDL support so I can see removing outer layer; do Joysticks still work after this change? (I don't have one to test with.)

Reply:
When reading this and having a look around, I asked myself if this `HAVE_SDL` ifdeffing is complete and/or necessary. You answered that here, so thanks. Would cleaning (removing) all those be wanted ? Worthwhile ?

Reply:
@BenjamenMeyer My joystick only works **after** this change. 😄

Reply:
@vincele Some of the ifdeffing might still be necessary. But only the inner ifdef, in this case, I believe. (The one that disables joystick functionality specifically.) After pushing up these changes last night, I realized that I might need to put back the inner ifdef after all.

Reply:
I don't actually know the answer to the question about `HAVE_SDL` ifdeffing. That might still be needed as well. In theory, we can build the project without SDL; it's not marked as a required dependency in CMakeLists.txt. However, in practice, I don't know that the game does anything useful without SDL. So maybe we should make SDL required and take out the ifdefs.

Reply:
The game engine started out as something very flexible and open ended.
That's why you could run the game with SDL or some other alternative.
Today, this is a bad idea and it was a bad design decision back then.
We should limit options to reduce complexity. SDL is a good choice- we should remove the ifdefs and use it exclusively.

Reply:
Makes sense to me. @BenjamenMeyer ?

Reply:
I agree with @royfalk; we don't need a super amount of flexibility and could really stand the benefits of reducing the complexity so let's move SDL to a required dependency starting with 0.10.x (current master) and we can start cleaning up headers there.

@stephengtuggy thanks for filing #1064 was just about to myself.